TD Info Model fixes

sebastiankb opened the pull request:
fix securityDefinitions from #1324


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1334.html" title="Last updated on Jan 12, 2022, 3:51 PM UTC (fc83764)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1334/21ce84a...fc83764.html" title="Last updated on Jan 12, 2022, 3:51 PM UTC (fc83764)">Diff</a>

relu91 commented on index.html:
```diff
@@ -1201,15 +1201,15 @@ <h2>Core Vocabulary Definitions</h2>
           "MUST">MUST</em> contain the anyURI
-          <code>https://www.w3.org/2022/wot/td/v1.1</code> either
+          <code>https://www.w3.org/2019/wot/td/v1</code> either
           directly when of type <code>anyURI</code> or as first
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
           <span class="rfc2119-assertion" id=
           "td-context-ns-thing-optional">When <code>@context</code>
           is an <a href="#dfn-array" class="internalDFN"
           data-link-type="dfn">Array</a>, the anyURI
-          <code>https://www.w3.org/2022/wot/td/v1.1</code> <em class=
+          <code>https://www.w3.org/2019/wot/td/v1</code> <em class=
```
Are these changes intended? cause in https://github.com/w3c/wot-thing-description/pull/1333 there are the same changes but from 2019 to 2022 🤔

sebastiankb replied:
many thanks, I fixed it.

mlagally commented on index.html:
```diff
@@ -2039,7 +2051,7 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
                     0..*
                 </td>
-                <td>Point to a software implementation of the TD.</td>
+                <td>Point to the web app manifest of a web application which provides a user interface with which a user can interact with the Thing (also see [[APPMANIFEST]]).</td>
```
This is a significant modification of the previous text.

These changes introduce a specific format to describe Web applications, which restricts the UI components to WebApps. Is this intended? 

The referenced document is a working draft, seems not to be suitable for a normative reference.

These late changes are difficult to review, and the implications cannot be properly considered in the limited review time.

sebastiankb replied:
Please see #1324 

![image](https://user-images.githubusercontent.com/13832739/146944414-76a7818f-5d89-470a-a9a3-54faf0ea5d1f.png)

sebastiankb replied:
I reverted this as well and put this in a separate PR #1340

mlagally commented on index.html:
```diff
@@ -2395,7 +2407,13 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
                 parameters (e.g., <code>charset=utf-8</code>) for
-                the media type [<cite><a class="bibref" data-link-type="biblio" href="#bib-rfc2046" title="Multipurpose Internet Mail Extensions (MIME) Part Two: Media Types">RFC2046</a></cite>].</td><td>mandatory</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string"><code>string</code></a></td></tr></tbody></table></section>
+                the media type [<cite><a class="bibref" data-link-type="biblio" href="#bib-rfc2046" title="Multipurpose Internet Mail Extensions (MIME) Part Two: Media Types">RFC2046</a></cite>].</td><td>mandatory</td><td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#string"><code>string</code></a></td></tr>
+<tr class="rfc2119-table-assertion" id="td-vocab-schema--ExpectedResponse"><td><code>schema</code></td><td>Used to define the output data schema 
```
Please clarify why this normative assertion is added.

mlagally replied:
These changes introduce new normative behavior and may have undesired impact.
They should be reviewed and discussed in the group before they are added.

sebastiankb replied:
see #1324 and https://github.com/w3c/wot-thing-description/issues/1053#issuecomment-959452562

sebastiankb replied:
this is also reverted and outsourced in a separate PR #1341

mlagally commented on validation/td-validation.ttl:
```diff
@@ -1082,7 +1094,7 @@
                     0..*
                 </td>
-                <td>Point to a software implementation of the TD.</td>
+                <td>Point to the web app manifest of a web application which provides a user interface with which a user can interact with the Thing (also see [[APPMANIFEST]]).</td>
```
See previous comment. This introduces a new specification, which is a working draft.

sebastiankb replied:
is reverted. see comment above

mlagally reviewed (changes requested):
This merge request contains additional normative assertions.

These come very late, and Oracle needs more time to properly review them and consider the implications.

This MR should not be merged without careful review and further discussion.

sebastiankb commented on index.html:
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
+           A 1.0 consumer can only be expected to parse a Thing Description which includes 
+           the 1.0 context, but can not be expected to parse a Thing Description which 
+           only has a 1.1 context. A Thing Description which only contains 
+           the 1.1 context should be a valid 1.1 Thing Description.
```
no, just a note.

sebastiankb commented on index.html:
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
+          The TD 1.1 context may exceptionally be placed as the second <a href="#dfn-array" class="internalDFN"
+          data-link-type="dfn">Array</a> entry if the TD 1.0 context must be specified as the first entry to allow 
```
see #1324 and https://github.com/w3c/wot-thing-description/issues/1243#issuecomment-949600882

sebastiankb commented on index.html:
```diff
@@ -1189,22 +1189,33 @@ <h2>Core Vocabulary Definitions</h2>
           directly when of type <code>anyURI</code> or as first
           element when of type <a href="#dfn-array" class=
           "internalDFN" data-link-type="dfn">Array</a>.</span>
+          The TD 1.1 context may exceptionally be placed as the second <a href="#dfn-array" class="internalDFN"
```
This should help to make clear that a TD that follows this specification should have also the TD 1.0 context if TD 1.0 processor are used. As next there is also a note about this situation.

sebastiankb replied:
this text and the note is removed again and a separate PR was provided #1339

sebastiankb commented:
@mlagally many thanks for reviewing this PR. This PR addresses the feedback given in #1324.

I outsourced your related comments in separate PRs #1341, #1340 and #1339 which we can be discussed and decided in tomorrow's call.

sebastiankb commented:
from today's TD call, decided to merge
